### ISSUE TITLE
Shipping estimator "blows" currently-set shipping address

### ISSUE THREAD

**Issue by lat9** [zencart/zencart#2234]
I've been chasing an issue for days trying to get One-Page Checkout readied.  Essentially, if a customer starts the checkout process (supplying a different shipping than billing address) and then (via the `shopping_cart` page) edits their order's contents and re-enters the checkout, their previously-set shipping address has been reset by the shipping-estimator's processing.

FWIW, I've verified that the shipping-estimator's processing also affects the 3-page checkout.  If you've chosen a different shipping-address than the customer's default address and click back to edit the shopping cart, the ship-to address has been (silently) reverted to the customer's default address.  This could lead to customer confusion, since I'm not sure that a customer will recognize that the address has been changed.

**Comment by lat9** [comment]
Noting also that, of course, there are no notifications to grab to circumvent that unwanted processing.

**Comment by mc12345678** [comment]
Need to check for `$_SESSION['sendto']` before checking `$_SESSION['cart_address_id']` as in:

````diff
53  if (!empty($_SESSION['customer_id'])) {
54    // user is logged in
55    if (isset($_POST['address_id'])){
56      // user changed address
57      $sendto = $_POST['address_id'];
+     }elseif (!empty($_SESSION['sendto'])) {
+       $sendto = $_SESSION['sendto'];
58    }elseif (!empty($_SESSION['cart_address_id'])){
````
Though possibly also want to refactor some of the lines to provide proper spacing as well.

Having looked into this today, seems that I need to modify a [previous PR](https://github.com/zencart/zencart/pull/1952/files) based on seeing system response when not logged in and updating after choosing a country and not having a zip code entered.

**Comment by drbyte** [comment]
Closed by #2286
